pkg/datarepair/checker/checker.go

```diff
@@ -86,11 +90,7 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 	err = checker.pointerdb.Iterate("", "", true, false,
 		func(it storage.Iterator) error {
 			var item storage.ListItem
-			lim := checker.limit
-			if lim <= 0 || lim > storage.LookupLimit {
-				lim = storage.LookupLimit
-			}
-			for ; lim > 0 && it.Next(&item); lim-- {
+			for it.Next(&item) {
```

Comment:
Iterate may lock the whole database at the moment.

Reply:
@egonelbre I've been looking for where a DB might lock, but can't find it in our code.  Can you point me to this?

Reply:
https://github.com/storj/storj/blob/master/satellite/satellitedb/locked.go

Reply:
https://github.com/storj/storj/blob/master/storage/boltdb/client.go#L107

Reply:
I read it as View() is ok... `Bolt allows only one read-write transaction at a time but allows as many read-only transactions as you want at a time.` [https://github.com/boltdb/bolt#transactions]

Reply:
I may have misremembered this or it may has changed at some point.

Reply:
It seems that there may be some issues in Bolt when updating inside a iterator, but that doesn't seem to apply anywhere in our codebase at the moment